### PR TITLE
Add wait helper documentation to component testing guide

### PR DESCRIPTION
**Pull request by toddjordan**
Fix to issue https://github.com/emberjs/guides/issues/909

I created an example I thought might be fairly common.  I would use an ajax example but I didn't want to complicate things by using something like pretender in the guides.  Am open to suggestions though.

### CODE REVIEW

File: source/testing/testing-components.md [hunk 1]
```diff
@@ -273,3 +273,71 @@ test('should change displayed location when current location changes', function
   assert.equal(this.$().text().trim(), 'You currently are located in Beijing, China', 'location display should change');
 });
 ```
+
+### Waiting on Asynchronous Behavior
+Often, interacting with a component will cause asynchronous behavior to occur, such as HTTP requests, or timers.  The
+`wait` helper is designed to handle these these scenarios, by providing a hook to ensure assertions are made after
+all ajax requests and timers are complete.
+
+Imagine you have a typahead component that uses `Ember.run.debounce` to limit requests to the server, and you want
```
Comment (michaelrkn) [inline source/testing/testing-components.md]: link to http://emberjs.com/api/classes/Ember.run.html#method_debounce

File: source/testing/testing-components.md [hunk 2]
```diff
@@ -273,3 +273,71 @@ test('should change displayed location when current location changes', function
   assert.equal(this.$().text().trim(), 'You currently are located in Beijing, China', 'location display should change');
 });
 ```
+
+### Waiting on Asynchronous Behavior
+Often, interacting with a component will cause asynchronous behavior to occur, such as HTTP requests, or timers.  The
+`wait` helper is designed to handle these these scenarios, by providing a hook to ensure assertions are made after
```
Comment (michaelrkn) [inline source/testing/testing-components.md]: add link to http://emberjs.com/api/classes/Ember.Test.html#method_wait

File: source/testing/testing-components.md [hunk 3]
```diff
@@ -273,3 +273,71 @@ test('should change displayed location when current location changes', function
...
+  this.$('input').val('test');
+  this.$('input').trigger('keyup');
+
+  return wait().then(() => {
+    assert.equal(this.$('.result').length, 2, 'two results rendered');
+  });
+
+});
+```
+
```
Comment (michaelrkn) [inline source/testing/testing-components.md]: extra linebreak

File: source/testing/testing-components.md [hunk 4]
```diff
@@ -273,3 +273,71 @@ test('should change displayed location when current location changes', function
...
+to verify that results are displayed after typing a character.
+
+> You can follow along by generating your own component with `ember generate
+> component delayed-typeahead`.
+
+```app/components/delayed-typeahead.js
+export default Ember.Component.extend({
+  actions: {
+    handleTyping() {
+      Ember.run.debounce(this, this.get('fetchResults'), this.get('searchValue'), 250);
```
Comment (michaelrkn) [inline source/testing/testing-components.md]: I think it make it clearer to explain what `fetchResults` does here.

**Comment by michaelrkn** [comment]
This looks great! I added a couple comments. @locks can you look over it as well?

File: source/testing/testing-components.md (hunk 4 above)
Comment (toddjordan) [inline source/testing/testing-components.md]: sure.  I'll add a comment line above the debounce call.

File: source/testing/testing-components.md (hunk 2 above)
Comment (toddjordan) [inline source/testing/testing-components.md]: I'll also drop the double 'these' :-P

**Pull request merged**